reactive/etcd.py

```diff
@@ -341,6 +392,16 @@ def passive_dismiss_context(cluster):
     cluster.dismiss()
 
 
+def close_open_ports():
+    ''' Close the previous port and open the port from configuration. '''
+    configuration = hookenv.config()
+    previous_port = configuration.previous('port')
+    port = configuration.get('port')
+    if previous_port != port:
+        close_port(previous_port)
```

Comment:
You had made mention this was of concern and its a legitimate concern. On first run it is indeed None.

subprocess.CalledProcessError: Command '['close-port', 'None/TCP']' returned non-zero exit status 2

This can be avoided with a quick fix to the inline if statement:

```python
if previous_port != port and previous_port is not None:
```